pennylane/ops/qubit/observables.py

```diff
@@ -715,10 +739,29 @@ def compute_diagonalizing_gates(
         # Adapting the approach discussed in the link below to work with arbitrary complex-valued state vectors.
         # Alternatively, we could take the adjoint of the Mottonen decomposition for the state vector.
         # https://quantumcomputing.stackexchange.com/questions/10239/how-can-i-fill-a-unitary-knowing-only-its-first-column
-        phase = qml.math.exp(-1j * qml.math.angle(state_vector[0]))
+
+        if qml.math.get_interface(state_vector) == "tensorflow":
+            dtype_name = qml.math.get_dtype_name(state_vector)
+            if dtype_name == "int32":
+                state_vector = qml.math.cast(state_vector, np.complex64)
+            elif dtype_name == "int64":
+                state_vector = qml.math.cast(state_vector, np.complex128)
+
+        angle = qml.math.angle(state_vector[0])
+        if qml.math.get_interface(angle) == "tensorflow":
+            if qml.math.get_dtype_name(angle) == "float32":
+                angle = qml.math.cast(angle, np.complex64)
+            else:
+                angle = qml.math.cast(angle, np.complex128)
+
+        phase = qml.math.exp(-1.0j * angle)
         psi = phase * state_vector
         denominator = qml.math.sqrt(2 + 2 * psi[0])
-        psi = qml.math.set_index(psi, 0, psi[0] + 1)  # psi[0] += 1, but JAX-JIT compatible
+        summed_array = np.zeros(qml.math.shape(psi), dtype=qml.math.get_dtype_name(psi))
+        summed_array[0] = 1.0
+        psi = (
+            psi + summed_array
+        )  # qml.math.set_index(psi, 0, psi[0] + 1)  # psi[0] += 1, but JAX-JIT compatible
```

Comment:
Doesn't that work?
```suggestion
        psi = qml.math.set_index(psi, 0, psi[0] + 1, like=qml.math.get_interface(psi))  # psi[0] += 1, but JAX-JIT compatible
```

Reply:
`set_index` is does not work with tensorflow.

Potentially I could dispatch on the type being tensorflow and only following this other type of logic in that case.

Reply:
I see, I thought JAX was the problem.